pkg/apis/core/validation/validation.go

```diff
@@ -2930,6 +2931,17 @@ func validateReadinessGates(readinessGates []core.PodReadinessGate, fldPath *fie
 	return allErrs
 }
 
+// Validate domain name, note this is a looser check than ValidateDNS1123Subdomain
+// which does ensure valid DNS names but is more restrictive than DNS.
+func validateDomainName(value string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	_, d := dns.IsDomainName(value)
+	if !d {
+		allErrs = append(allErrs, field.Invalid(fldPath, value, fmt.Sprintf("not a valid domain name: %s", value)))
```

Comment:
For the sake of errors, I think it would be easier to write the validation ourselves.

test length
split on "."
test each length